plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -920,7 +921,7 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
 
```

Comment:
You should use the region even if endpoint isn't set
```
        String region = hadoopConfig.get(S3_REGION);
        if (region != null) {
            clientBuilder.setRegion(region);
            regionOrEndpointSet = true;
        }

        String endpoint = hadoopConfig.get(S3_ENDPOINT);
        if (endpoint != null) {
            clientBuilder.setEndpointConfiguration(new EndpointConfiguration(endpoint, region));
            regionOrEndpointSet = true;
        }
```

Reply:
You are actually right. I totally ignored the fact that setting `regionOrEndpointSet` to true will prevent `US_EAST_1` to be set as a default few lines later.
It will not effectively change anything as `US_EAST_1` is always chosen as a default one, but we should set it nonetheless.

Made the change and altered tests.